tooling/noir_js/src/program.ts

```diff
@@ -1,13 +1,24 @@
 /* eslint-disable  @typescript-eslint/no-explicit-any */
 import { Backend, CompiledCircuit } from '@noir-lang/types';
 import { generateWitness } from './witness_generation.js';
+import initAbi from '@noir-lang/noirc_abi';
+import initACVM from '@noir-lang/acvm_js';
 
 export class Noir {
   constructor(
     private circuit: CompiledCircuit,
     private backend: Backend,
   ) {}
 
+  async init(): Promise<void> {
+    // If these are available, then we are in the
+    // web environment. For the node environment, this
+    // is a no-op.
+    if (typeof initAbi === 'function') {
+      await Promise.all([initAbi(), initACVM()]);
+    }
```

Comment:
Add `await this.backend.init()`?

Reply:
I made an issue for this, If we do, then should we remove the lazy initting we are doing in the backend?

Reply:
ie we add .init() to the backend interface, then we only ever init once inside of the Noir class

Reply:
See comment here: https://github.com/noir-lang/noir/issues/2910#issuecomment-1742811237